### PR TITLE
Refactor CalendarWidget disabled/loading props

### PR DESCRIPTION
**Pull request by narin**
## Description
Refactor the way the CalendarWidget handles disabled (loading) state.  The calendar widget does not do any data fetching itself, so we'll replace the loading props with disabled props, and have the parent component determine whether to disable the widget or not and pass an appropriate message.  This PR:

- Removes the loadingStatus and loadingErrorMessage props
- Adds new disabled and disabledMessage props
   - When disabled is true, you should get the UI that shows now for the loading state
   - Instead of a LoadingIndicator, use the value from disabledMessage for the message
- In DateTimeSelectPage, handle loading errors by display an error message instead of the calendar.
- Fix a bug where a user was able to perform click events even when the loading overlay was visible

## Testing done
Local and unit

## Screenshots
image

### CODE REVIEW

File: src/applications/vaos/new-appointment/sass/_calendar.scss [hunk 1]
```diff
@@ -1,8 +1,14 @@
-.vaos-calendar__loading {
+.vaos-calendar__disabled {
   opacity: 0.3;
+  pointer-events: none;
+  background: none !important;
+  -webkit-user-select: none;  /* Chrome all / Safari all */
+  -moz-user-select: none;     /* Firefox all */
+  -ms-user-select: none;      /* IE 10+ */
+  user-select: none;          /* Likely future */
```
Comment (narin) [inline src/applications/vaos/new-appointment/sass/_calendar.scss:8]: Disables click events for elements behind the disabled overlay

Comment (jbalboni) [reply src/applications/vaos/new-appointment/sass/_calendar.scss:8]: Does it disable keyboard nav? I noticed that the other day with the current overlay, you can tab through the calendar still.

We also may want to consider how screen reader users are being told the calendar is disabled, though that's sort of similar to the keybaord nav question.

**Review by jbalboni — APPROVED** [review]
Changes look good, but have one question

**Comment by narin** [comment]
@jbalboni I've also passed on the disabled prop to the next/prev buttons and calendar cells so tabbing through these will be disabled when in the loading state.

**Pull request merged**